src/filters/array.rs

```diff
@@ -0,0 +1,217 @@
+use liquid_value::Value;
+
+use super::{check_args_len, invalid_input};
+use interpreter::FilterResult;
+use std::fmt::Write;
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the given value appended at the end.
+pub fn push(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 1, 0)?;
+
+    let value = args[0].clone();
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+    array.push(value);
+
+    Ok(Value::array(array))
+}
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the last element removed.
+pub fn pop(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 0, 0)?;
+
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+    array.pop();
+
+    Ok(Value::array(array))
+}
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the given value appended at the beginning.
+pub fn unshift(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 1, 0)?;
+
+    let value = args[0].clone();
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+    array.insert(0, value);
+
+    Ok(Value::array(array))
+}
+
+/// Receives a `Value::Array` as an input.
+/// Returns a copy of the input with the first element removed.
+pub fn shift(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 0, 0)?;
+
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .clone();
+
+    if !array.is_empty() {
+        array.remove(0);
+    }
+
+    Ok(Value::array(array))
+}
+
+/// Convert an array into a sentence.
+pub fn array_to_sentence_string(input: &Value, args: &[Value]) -> FilterResult {
+    check_args_len(args, 0, 1)?;
+
+    let connector = args
+        .get(0)
+        .and_then(|arg| Some(arg.to_string()))
+        .unwrap_or_else(|| "and".to_string());
+
+    let mut array = input
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .iter();
+
+    let mut sentence = array
+        .next()
+        .and_then(|v| Some(v.to_string()))
+        .unwrap_or_else(|| "".to_string());
+
+    let last = array.next_back();
+
+    for value in array {
+        write!(sentence, ", {}", value).unwrap();
```

Comment:
> I'm just unsure if unwraping write! is the way to go. As far as I'm aware, it should be safe to write to a string but if I'm mistaken I can make it return an error instead.

I think you are correct.  The only failable operation should be a allocation (no IO to fail, formatting is compile-time) and allocations are panics.

I think I would prefer `expect`s though.  It documents your assumption to help people when auditing to consider whether we are right or not.